### ISSUE TITLE
ClickHouse keeps gracefully shutting down - Running as a Docker service 

### ISSUE THREAD

**Issue by DavidTheProgrammer** [ClickHouse/ClickHouse#63332]
**Describe the unexpected behaviour**
ClickHouse worked for about 3 months just fine but now it keeps shutting down gracefully with no obvious errors or causes of failure present in both the access logs and error logs. As additional information, I'm running it as a single service on a docker swarm.

Here are some configurations I've tried hoping it helps.

This is the config I've added to the `/etc/clickhouse-server/config.d/` directory. `clickhouse-config.xml`
```xml
<clickhouse>
    <!-- Limit server Max RAM usage to a ratio of physical RAM on machine -->
    <max_server_memory_usage_to_ram_ratio>0.5</max_server_memory_usage_to_ram_ratio>

    <merge_tree>
        <merge_max_block_size>1024</merge_max_block_size>
    </merge_tree>
</clickhouse>
```

This is the query config I've added to `/etc/clickhouse-server/users.d/` directory `clickhouse-query-config.xml`
```xml
<clickhouse>
    <profiles>
        <default>
            <max_memory_usage>16000000000</max_memory_usage><!-- 16GB -->
            <max_memory_usage_for_all_queries>4000000000</max_memory_usage_for_all_queries><!-- 4GB -->
            <max_memory_usage_for_user>4000000000</max_memory_usage_for_user><!-- 4GB -->
            <max_bytes_before_external_group_by>2000000000</max_bytes_before_external_group_by><!-- 2GB -->
            <max_bytes_before_external_sort>2000000000</max_bytes_before_external_sort><!-- 2GB -->
        </default>
    </profiles>
</clickhouse>
```

I've also attached both files hoping someone can see what's gone wrong.

[clickhouse-server.err.log](https://github.com/ClickHouse/ClickHouse/files/15200221/clickhouse-server.err.log)
[clickhouse-server.log](https://github.com/ClickHouse/ClickHouse/files/15200222/clickhouse-server.log)

**Comment by Algunenano** [comment]
Something is telling the process to stop:
```
2024.05.03 12:33:42.990041 [ 47 ] {} <Trace> BaseDaemon: Received signal 15
2024.05.03 12:33:42.990102 [ 47 ] {} <Information> Application: Received termination signal (Terminated)
2024.05.03 12:33:42.990119 [ 1 ] {} <Debug> Application: Received termination signal.
2024.05.03 12:33:42.990194 [ 1 ] {} <Debug> Application: Waiting for current connections to close.
```

Review you docker / k8s / whatever config because there is something that shuts down the service

**Comment by DavidTheProgrammer** [comment]
So I looked into this and scoured the internet. I couldn't find anything related to this and how I was running the service. However, I think I found out where the SIGTERM command to shutdown the container was coming from, might have been docker itself! I added a healthcheck in compose file and I think with time the container couldn't start up fast enough and so docker would kill it and start up a new one. I specifically added a limit of 5 retries and so all the containers would just be killed and it would stop after the 5th attempt. 

Fix: Removed the healthcheck entirely from the compose file and now the DB starts up with no issues. 

This is a non-issue.
